src/tests/ftest/util/server_utils.py

```diff
@@ -241,10 +238,24 @@ def run_server(test, hostfile, setname, uri_path=None, env_dict=None,
             [line.split(' ')[0] for line in genio.read_all_lines(hostfile)])
         server_count = len(servers)
 
+        # Pile of build time variables
+        with open("../../../.build_vars.json") as json_vars:
+        # TBD with open(os.path.join(test.basepath, ".build_vars.json")) as json_vars:
+            build_vars = json.load(json_vars)
+
         # Create the DAOS server configuration yaml file to pass
         # with daos_server -o <FILE_NAME>
         print("Creating the server yaml file")
-        server_yaml = os.path.join(test.basepath, AVOCADO_FILE)
+
+        # set default shared dir for daos tests in case DAOS_TEST_SHARED_DIR
+        # is not set, for RPM env and non-RPM env.
+        if build_vars["PREFIX"] == "/usr":
+            default_yamldir = os.path.expanduser('~/daos_test')
+        else:
+            default_yamldir = build_vars["PREFIX"]
+
+        yaml_prefix = os.getenv('DAOS_TEST_SHARED_DIR', default_yamldir)
```

Comment:
The `test` argument passed into this function is a `TestWithServers` object, so you can just use `test.tmp` in place of `default_yamldir`. e.g.
```
yaml_prefix = os.getenv('DAOS_TEST_SHARED_DIR', test.tmp)
```
or even the following if the assignment of `self.tmp` included the optioanl `DAOS_TEST_SHARED_DIR` env handling in the `TestWithServers.setUp()` method:
```
server_yaml = os.path.join(test.tmp, AVOCADO_FILE)
```

Reply:
Got this error after using test.tmp:
08:44:43 DEBUG|   File "/var/lib/jenkins/jenkins-1/docker_1/workspace/daos-stack_daos_PR-1048/install/lib/daos/TESTING/ftest/util/server_utils.py", line 248, in run_server
08:44:43 DEBUG|     server_yaml = os.path.join(test.tmp, AVOCADO_FILE)
08:44:43 DEBUG|   File "/usr/lib64/python2.7/posixpath.py", line 77, in join
08:44:43 DEBUG|     elif path == '' or path.endswith('/'):
08:44:43 DEBUG| AttributeError: 'NoneType' object has no attribute 'endswith'

Reply:
Is test.tmp the same as self.tmp set in test.py?
        if self.prefix != "/usr":
131             self.tmp = os.path.join(self.prefix, 'tmp')
132         else:
133             self.tmp = os.getenv('DAOS_TEST_SHARED_DIR', \
134                                  os.path.expanduser('~/daos_test'))
135             os.makedirs(self.tmp, exist_ok = True)